philly-js-club/philly-js-club-website
# Add RSS feed, resolves 38

## PR Checklist

- [x] Addresses an existing open issue: fixes #038
- [x] That issue was marked as [`status: accepting prs`](https://github.com/philly-js-club/philly-js-club-website-remix/issues?q=is%3Aopen+is%3Aissue+label%3A%22status%3A+accepting+prs%22)
- [x] Steps in [CONTRIBUTING.md](https://github.com/philly-js-club/philly-js-club-website-remix/blob/main/.github/CONTRIBUTING.md) were taken

## Overview



Added an RSS feed generator and RSS generator config. This will generate an RSS.xml file in the public folder.  
Infomation about RSS https://validator.w3.org/feed/docs/rss2.html#whatIsRss

### How to test: 
- From the root, run `node lib/rss/rss-gen.js`
- Go to the public folder and find RSS.xml
- You can go to `https://validator.w3.org/feed/check.cgi` to validate the RSS

### How to update RSS XML feed: 

Feed items can be updated in the `feed` section in `lib/rss/rss.config.json`

- Tip: Please don't update `XMLOptions` and/or `RSSOptions` their functionality has not been expanded

### What next: 
- Where do we put the XML file? (I placed the XML in the public folder for now)
- When is the XML file generated? (This can be on build or rss.config.json changes)
- Do we add the XML file to the .gitignore? (This is dependent on how it is generated)
- Add RSS XML ref to the site

### Where it can expand
- Expand `XMLOptions` and `RSSOptions` functionality 
- Tag Validation
- Error handling
- XML validation
- atom implementation

### Sample
```
<?xml version="1.0" ?><rss version="2.0"><channel><description>The Philadelphia JavaScript Club, founded in 2022, is a a place for developers of all skill and experience levels to connect, show off their work, and learn new things.</description><language>en-US</language><link>https://phillyjs.com/</link><title>Philly JS Club</title><item><description>The Philadelphia JavaScript Club, founded in 2022, is a a place for developers of all skill and experience levels to connect, show off their work, and learn new things.</description><link>https://phillyjs.com/about</link><title>Welcome to Philly JS club</title></item></channel></rss>
